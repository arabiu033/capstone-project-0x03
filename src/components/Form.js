import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { fetchLaureates } from '../redux/space/laureatesSlice';

let fetch = true;
const Form = () => {
  const [filter, setFilter] = useState({
    id: 'che', count: 50, to: 1951, from: 1901,
  });
  const dispatch = useDispatch();
  useEffect(() => {
    if (fetch) {
      dispatch(fetchLaureates(filter));
      fetch = false;
    }
  });
  const handleChange = (e) => {
    setFilter((data) => ({ ...data, [e.target.name]: e.target.value }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (+filter.to > 2022) return;
    if (+filter.from < 1901) return;
    dispatch(fetchLaureates(filter));
  };
  return (
    <form name="filte" onSubmit={handleSubmit}>
      <label htmlFor="year">
        <span>Category </span>
        <select name="id" id="category" onChange={handleChange}>
          <option value="phy">Physics</option>
          <option value="che" defaultValue={filter.id}>Chemistry</option>
          <option value="eco">Economics</option>
          <option value="med">Medicine</option>
          <option value="Pea">Peace</option>
          <option value="lit">Literature</option>
        </select>
      </label>
      <label htmlFor="year">
        <span>Year From (min 1901) </span>
        <input type="text" id="" value={filter.from} name="from" onChange={handleChange} />
      </label>
      <label htmlFor="year">
        <span>Year To (max 2022) </span>
        <input type="text" id="" value={filter.to} name="to" onChange={handleChange} />
      </label>
      <button type="submit">Filtered</button>
    </form>
  );
};

export default Form;
