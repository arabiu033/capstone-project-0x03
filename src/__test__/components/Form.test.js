import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';

import Form from '../../components/Form';

test('renders correctly Form component', () => {
  const form = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <Form />
      </Provider>
    </BrowserRouter>
  ).toJSON();
  expect(form).toMatchSnapshot();
});