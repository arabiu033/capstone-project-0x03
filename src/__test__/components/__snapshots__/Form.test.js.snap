// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Form component 1`] = `
<form
  name="filte"
  onSubmit={[Function]}
>
  <label
    htmlFor="year"
  >
    <span>
      Category 
    </span>
    <select
      id="category"
      name="id"
      onChange={[Function]}
    >
      <option
        value="phy"
      >
        Physics
      </option>
      <option
        defaultValue="che"
        value="che"
      >
        Chemistry
      </option>
      <option
        value="eco"
      >
        Economics
      </option>
      <option
        value="med"
      >
        Medicine
      </option>
      <option
        value="Pea"
      >
        Peace
      </option>
      <option
        value="lit"
      >
        Literature
      </option>
    </select>
  </label>
  <label
    htmlFor="year"
  >
    <span>
      Year From (min 1901) 
    </span>
    <input
      id=""
      name="from"
      onChange={[Function]}
      type="text"
      value={1901}
    />
  </label>
  <label
    htmlFor="year"
  >
    <span>
      Year To (max 2022) 
    </span>
    <input
      id=""
      name="to"
      onChange={[Function]}
      type="text"
      value={1951}
    />
  </label>
  <button
    type="submit"
  >
    Filtered
  </button>
</form>
`;
